src/react/providers/room-provider.tsx

```diff
@@ -0,0 +1,85 @@
+import { RoomOptions, RoomOptionsDefaults } from '@ably/chat';
+import React, { ReactNode, useEffect, useMemo } from 'react';
+
+import { RoomContext } from '../contexts/room-context.js';
+import { useChatClient } from '../hooks/use-chat-client.js';
+
+/**
+ * Props for the RoomProvider component.
```

Comment:
Could we wrap the RoomProvider in {@link RoomProvider} so it links correctly in the docs please? :)